packages/date/src/index.js

```diff
@@ -136,7 +135,13 @@ export function getRangeLabel( after, before ) {
  * @return {DateValue} -  DateValue data about the selected period
  */
 export function getLastPeriod( period, compare ) {
-	const primaryStart = moment().startOf( period ).subtract( 1, period );
+	const primaryStart = moment(
+		moment()
+			.tz( window.wcSettings ? window.wcSettings.timeZone : null )
+			.format( 'YYYY-MM-DD HH:mm:ss' )
+	)
+		.startOf( period )
+		.subtract( 1, period );
```

Comment:
Is the wrapping of the moment tz call necessary? could we just do:
`moment().tz( window.wcSettings ? window.wcSettings.timeZone : null ).startOf( period ).subtract( 1, period )` ?

Although it appears (when trying locally) that `moment().tz(null)` returns undefined, so maybe we want to make that more explicit, could create a helper function for that (given it is the same logic in the `getCurrentPeriod` method).

Reply:
>Is the wrapping of the moment tz call necessary?

I think it my initial attempts it was since the new moment would include timezone data, but now that we use that moment when appending timestamps, I think we're good to directly use `.tz()`.

>could create a helper function 

Good idea.  Added a helper `getStoreTimeZoneMoment` and a constant `defaultDateTimeFormat`.